Common pattern for storage clusters for NFS, BeeGFS and GlusterFS

oneswig opened the pull request:
Use a transient bastion in a similar manner to other deployments.
Add storage tags for GlusterFS and BeeGFS.
Needs merging with embedded metadata for client access. (https://github.com/cedadev/jasmin-appliances/pull/16)

Commit:
allow for beegfs cluster to be built independently (with fip)

Commit:
deploy storage clusters independently with ephemeral bastion

Commit:
first working version to build gluster-cluster using stackhpc.gluster-cluster role

oneswig committed:
Add storage tags and ephemeral bastion for BeeGFS deployment.

oneswig committed:
Update NFS playbook to use transient bastion, tags, etc.

oneswig committed:
Remove bastion host config in favour of group_vars definition.

oneswig committed:
Gluster support rename

oneswig committed:
Further work consolidating Gluster server and client

oneswig committed:
NFS setup may not have the server in the current inventory (WIP)

brtkwr commented on beegfs-infra.yml:
```diff
@@ -0,0 +1,51 @@
+      cluster_network: "caastest-U-internal"
+      cluster_net: "{{ beegfs_net }}"
+      cluster_name: "{{ beegfs_name }}"
+      #cluster_groups: "{{ [cluster_gw | combine({'num_nodes': 0})] + beegfs_groups }}"
```
remove

brtkwr commented on beegfs-infra.yml:
```diff
@@ -0,0 +1,51 @@
+      cluster_groups: "{{ beegfs_groups + [cluster_gw] }}"
+      cluster_deploy_user: "{{ beegfs_deploy_user }}"
+      cluster_tag: "{{ beegfs_tag }}"
+      #cluster_gw_group: "{{ cluster_gw_group }}"
```
remove

brtkwr commented on beegfs-infra.yml:
```diff
@@ -0,0 +1,51 @@
+      cluster_tag: "{{ beegfs_tag }}"
+      cluster_stack_update_only: true
+
+  # Hack pending outcome of https://github.com/ansible/ansible/pull/53757
```
and https://review.openstack.org/643195

brtkwr commented on gluster-infra.yml:
```diff
@@ -0,0 +1,56 @@
+      cluster_tag: "{{ gluster_tag }}"
+      cluster_stack_update_only: true
+
+  # Hack pending outcome of https://github.com/ansible/ansible/pull/53757
```
and https://review.openstack.org/643195

brtkwr commented on nfs-infra.yml:
```diff
@@ -2,22 +2,48 @@
+      cluster_tag: "{{ nfs_tag }}"
+      cluster_stack_update_only: true
+
+  # Hack pending outcome of https://github.com/ansible/ansible/pull/53757
```
and https://review.openstack.org/643195

brtkwr reviewed (changes requested):
Mostly nitpickings, otherwise lgtm

brtkwr commented on beegfs-infra.yml:
```diff
@@ -0,0 +1,51 @@
+      cluster_auth: "{{ openstack_auth }}"
+      cluster_network: "caastest-U-internal"
+      cluster_net: "{{ beegfs_net }}"
+      cluster_name: "{{ beegfs_name }}"
```
This is going to be problematic as the cluster_name gets overridden at a global scope.

brtkwr commented on beegfs-infra.yml:
```diff
@@ -0,0 +1,51 @@
+      cluster_auth: "{{ openstack_auth }}"
+      cluster_network: "caastest-U-internal"
+      cluster_net: "{{ beegfs_net }}"
+      cluster_name: "{{ beegfs_name }}"
```
This is going to be problematic as the cluster_name gets overridden at a global scope.

brtkwr commented on beegfs-infra.yml:
```diff
@@ -0,0 +1,51 @@
+    - role: jasmin.cluster-infra
+      cluster_auth_type: "{{ openstack_auth_type }}"
+      cluster_auth: "{{ openstack_auth }}"
+      cluster_network: "caastest-U-internal"
```
Probably shouldn't hard code this in here...

brtkwr commented on beegfs-infra.yml:
```diff
@@ -0,0 +1,51 @@
+    - role: jasmin.cluster-infra
+      cluster_auth_type: "{{ openstack_auth_type }}"
+      cluster_auth: "{{ openstack_auth }}"
+      cluster_network: "caastest-U-internal"
```
Remove

oneswig committed:
Feedback from @brtknr

brtkwr commented on beegfs.yml:
```diff
@@ -2,7 +2,9 @@
+    -  beegfs_clients
   roles:
     # Note: The NFS clients are likely to be drawn from across a multi-node
     # platform deployment. This playbook assumes the existence of a meta-group,
```
Are these comments relevant on a Beegfs playbook?

brtkwr commented on beegfs-infra.yml:
```diff
@@ -0,0 +1,51 @@
+  roles:
+    - role: jasmin.cluster-infra
+      cluster_auth_type: "{{ openstack_auth_type }}"
+      cluster_auth: "{{ openstack_auth }}"
```
these auth config are injected by awx... not sure if we need to explicitly pass them in here.

brtkwr commented on config/gluster.yml:
```diff
@@ -0,0 +1,23 @@
+
+gluster_deploy_user: 'centos'
+
+gluster_net: "caastest-U-internal"
```
remove

brtkwr commented on gluster-infra.yml:
```diff
@@ -0,0 +1,58 @@
+      cluster_auth_type: "{{ openstack_auth_type }}"
+      cluster_auth: "{{ openstack_auth }}"
+      cluster_network: "caastest-U-internal"
+      cluster_name: "{{ gluster_name }}"
```
remove

brtkwr commented on gluster-infra.yml:
```diff
@@ -0,0 +1,58 @@
+    - role: jasmin.cluster-infra
+      cluster_auth_type: "{{ openstack_auth_type }}"
+      cluster_auth: "{{ openstack_auth }}"
+      cluster_network: "caastest-U-internal"
```
remove

brtkwr commented on gluster-infra.yml:
```diff
@@ -0,0 +1,58 @@
+      cluster_auth_type: "{{ openstack_auth_type }}"
+      cluster_auth: "{{ openstack_auth }}"
+      cluster_network: "caastest-U-internal"
+      cluster_name: "{{ gluster_name }}"
```
remove

brtkwr commented on gluster-infra.yml:
```diff
@@ -0,0 +1,58 @@
+    - role: jasmin.cluster-infra
+      cluster_auth_type: "{{ openstack_auth_type }}"
+      cluster_auth: "{{ openstack_auth }}"
+      cluster_network: "caastest-U-internal"
```
remove

oneswig committed:
Use the interface parameter cluster_name for storage clusters.

oneswig committed:
User the interface parameter cluster_network for storage clusters.

oneswig committed:
Rearrange CaaS config to comply with the interface used with AWX.

Use group_vars for the openstack group for defining cluster platforms.
Update the README to define and document the extra parameters to supply.

oneswig committed:
Fix markdown table syntax

brtkwr commented on config/gluster.yml:
```diff
@@ -0,0 +1,17 @@
+---
+gluster_deploy_user: 'centos'
```
should these configs be removed now that they live under group vars?

Commit:
Remove brtknr's signature from the bottom

brtkwr commented:
Is it okay to rebase this over #23 as you've moved some of the variables aorund?

brtkwr commented on config/nfs.yml:
```diff
@@ -1,12 +1,6 @@
-# cluster node hostnames.
-nfs_name: nfs
-
 nfs_deploy_user: 'centos'
```
This can be removed too

oneswig committed:
Move the OpenStack auth config into extra vars.

oneswig committed:
Remove subsumed extra vars

brtkwr commented:
LGTM
